src/dbnode/storage/shard.go

```diff
@@ -2562,6 +2570,78 @@ func (s *dbShard) Repair(
 	return repairer.Repair(ctx, nsCtx, nsMeta, tr, s)
 }
 
+func (s *dbShard) AggregateTiles(
+	ctx context.Context,
+	reader fs.DataFileSetReader,
+	sourceNsID ident.ID,
+	sourceBlockStart time.Time,
+	sourceShard databaseShard,
+	opts AggregateTilesOptions,
+	wOpts series.WriteOptions,
+) (int64, error) {
+	latestSourceVolume, err := sourceShard.latestVolume(sourceBlockStart)
```

Comment:
Better to pass in source shard as uint32 and latest volume rather than pass in as `databaseShard` (less calling private methods, etc).

Reply:
Agreed, but I'll address this in one of the subsequent PRs as this part of code has diverged significantly (it is now handling multiple source shards as opposed to just one).

Reply:
*should have said "multiple source BLOCKS as opposed to just one"